### PR TITLE
[Refactor] Make TaskWorkerPool does not rely on AgentServer

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

### CODE REVIEW

File: be/src/agent/agent_server.cpp [hunk 1]
```diff
@@ -343,39 +340,4 @@ void AgentServer::publish_cluster_state(TAgentResult& t_agent_result, const TAge
     status.to_thrift(&t_agent_result.status);
 }
 
-void AgentServer::finish_task(const TFinishTaskRequest& finish_task_request) {
```
Comment (sduzh) [inline be/src/agent/agent_server.cpp:346]: Moved to `finish_task.cpp`

File: be/src/agent/agent_server.cpp [hunk 2]
```diff
@@ -343,39 +340,4 @@ void AgentServer::publish_cluster_state(TAgentResult& t_agent_result, const TAge
...
-        try_time += 1;
-        StarRocksMetrics::instance()->finish_task_requests_failed.increment(1);
-        LOG(WARNING) << "finish task failed retry: " << try_time << "/" << TASK_FINISH_MAX_RETRY
-                     << "client_status: " << client_status << " status_code: " << result.status.status_code;
-
-        sleep(sleep_time_second);
-    }
-}
-
-AgentStatus AgentServer::report_task(const TReportRequest& request, TMasterResult* result) {
```
Comment (sduzh) [inline be/src/agent/agent_server.cpp:377]: Moved to `report_task.cpp`

File: be/src/agent/finish_task.h [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+namespace starrocks {
+
+class TFinishTaskRequest;
+
+void finish_task(const TFinishTaskRequest& finish_task_request);
```
Comment (imay) [inline be/src/agent/finish_task.h:7]: This function name is too common, do we need to give it a namespace or make it a class static function?

Comment (sduzh) [reply be/src/agent/finish_task.h:7]: The function parameter types are part of the function signature,  no naming conflict.

**Review by imay — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Pull request merged**